src/pudl/scripts/dbt_helper.py

```diff
@@ -365,13 +366,37 @@ def update_row_counts(
 
     partition_expr = partition_expressions[0]  # TODO: support multiple partitions
     new = _calculate_row_counts(table_name, partition_expr)
-    combined = _combine_row_counts(filtered, new)
-    _write_row_counts(combined)
 
-    return UpdateResult(
-        success=True,
-        message=f"Successfully updated row count table with counts from {table_name}, partitioned by {partition_expr}.",
+    # Make old and new row counts comparable so we can detect changes
+    row_count_idx = ["table_name", "partition"]
+    std_old = (
+        old.astype("string")
+        .fillna(value="")
+        .sort_values(row_count_idx)
+        .reset_index(drop=True)
     )
+    std_new = (
+        new.astype("string")
+        .assign(partition=lambda df: df["partition"].replace("None", pd.NA))
+        .fillna(value="")
+        .sort_values(row_count_idx)
+        .reset_index(drop=True)
+    )
```

Comment:
The old and new row count dataframes comapre as different even when the contents are really the same, because of different NA values (`np.nan` vs. `None`) and the resulting different data types. I standardized them to do the comparison below, but maybe this expectation of comparability should be integrated further upstream?